### PR TITLE
Allow JMX scraping over SSL

### PR DESCRIPTION
**Pull request by msvab**
Hello @brian-brazil ,

This PR enables JMX scraping over SSL. If you have any issues with this, please let me know.

Cheers,
Michal

### CODE REVIEW

**Comment by brian-brazil** [comment]
How would a user configure this?

A trust store existing at the system level doesn't mean SSL is desired.

**Comment by msvab** [comment]
Good point, I didn't think there would be other things running in that JVM, but that's not necessarily true. I'll add a way to configure it.

**Comment by brian-brazil** [comment]
System properties can be configured at the machine level.

It's recommenced to use the jmx exporter as a java agent, it bypasses all this.

**Comment by msvab** [comment]
I've added a ssl option to the yaml config. I was also hoping to add configuration for the ssl keystore and truststore, but after looking through the JMX api I've discovered that there is no way to provide the keystore and truststore programmatically and so it has to be configured with system properties.

File: collector/src/main/java/io/prometheus/jmx/JmxScraper.java [hunk 1]
```diff
@@ -86,14 +91,19 @@ public void doScrape() throws Exception {
...
-          HashMap credential = null;
+          Map<String, Object> environment = new HashMap<String, Object>();
           if(username != null && username.length() != 0 && password != null && password.length() != 0) {
-            credential = new HashMap();
             String[] credent = new String[] {username, password};
-            credential.put(javax.management.remote.JMXConnector.CREDENTIALS, credent);
-          }       
+            environment.put(javax.management.remote.JMXConnector.CREDENTIALS, credent);
+          }
+          if(ssl) {
```
Comment (brian-brazil) [inline collector/src/main/java/io/prometheus/jmx/JmxScraper.java]: space before (

**Comment by brian-brazil** [comment]
This also needs docs

**Comment by msvab** [comment]
I've documented the new config option. Is this sufficient?

**Comment by brian-brazil** [comment]
Thanks!

**Comment by msvab** [comment]
Thanks for such a fast turnaround!

**Pull request merged**